### PR TITLE
Update 0.23-scala3 template

### PR DESCRIPTION
**Pull request by channingwalton**
I humbly offer this PR to update the 0.23-scala3 template with the latest libs as asked for in issue #303.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -8,7 +8,12 @@ ThisBuild / githubWorkflowBuild := Seq(
     name = Some("Test generated template")
   )
 )
-ThisBuild / githubWorkflowJavaVersions := Seq("adopt@1.8", "adopt@1.11", "adopt@1.15")
+ThisBuild / githubWorkflowJavaVersions := Seq(
+  JavaSpec(JavaSpec.Distribution.Adopt, "8"),
+  JavaSpec(JavaSpec.Distribution.Adopt, "11"),
+  JavaSpec(JavaSpec.Distribution.Adopt, "15"),
```
Comment (armanbilge) [inline build.sbt]: I propose we drop 15, it's not an LTS

File: src/main/g8/build.sbt [hunk 2]
```diff
@@ -8,7 +8,7 @@ lazy val root = (project in file("."))
     organization := "$organization$",
     name := "$name;format="norm"$",
     version := "0.0.1-SNAPSHOT",
-    scalaVersion := "3.1.0",
+    scalaVersion := "3.2.1",
```
Comment (armanbilge) [inline src/main/g8/build.sbt]:
```suggestion
    scalaVersion := "3.2.2",
```

File: src/main/g8/default.properties [hunk 3]
```diff
@@ -8,8 +8,8 @@ package = $organization$.$name;format="norm,word"$
 #http4s_version = maven(org.http4s, http4s-blaze-server_2.12, stable)
 #logback_version = maven(ch.qos.logback, logback-classic, stable)
 
-sbt_version = 1.5.5
-http4s_version = 0.23.6
-logback_version = 1.2.6
+sbt_version = 1.8.2
+http4s_version = 0.23.19-RC3
```
Comment (armanbilge) [inline src/main/g8/default.properties]:
```suggestion
http4s_version = 0.23.18
```

File: build.sbt (hunk 1 above)
Comment (danicheg) [reply build.sbt]: Agree on that one

Comment (channingwalton) [reply build.sbt]: Done

**Comment by armanbilge** [comment]
I think CI is too old to support JDK 17 😕 we need to switch to sbt-typelevel-github-actions.

https://typelevel.org/sbt-typelevel/gha.html

https://github.com/http4s/http4s.g8/blob/7c5f0c1e06e4b1595d33e1e1778ed6e00eba6e31/project/plugin.sbt#L2

**Comment by channingwalton** [comment]
> I think CI is too old to support JDK 17 😕 we need to switch to sbt-typelevel-github-actions.

Ah ok. I will fiddle with it.

**Comment by channingwalton** [comment]
Apologies, I'm still messing with GHA. I think its ok now, will wait for the build.

**Comment by channingwalton** [comment]
Hmm, this is quite painful. I'll see if I can get something working in a test repo.

**Comment by rossabaker** [comment]
Actions is in a semi-outage, which might be contributing to your pain.

Some of these upgrades are useful on the main branch.  I haven't been maintaining anything for Scala 3, and I _think_ the branches may have diverged too much to use the regular merging strategy, but I may investigate some cherry picks once the dust settles.

**Comment by rossabaker** [comment]
Close and reopen so the Actions run.

**Review by rossabaker — APPROVED** [review]

**Review by armanbilge — APPROVED** [review]
Nice job, and thank you for persisting, this became a mission-and-a-half 😅

**Comment by channingwalton** [comment]
Woot! Thanks, I learnt things.

**Comment by BobenHome** [comment]
Why sbt new http4s/http4s.g8 --branch 0.23-scala3  not success？
image

**Pull request merged**